### ISSUE TITLE
[Feature Request]: Provide a way to remove specific properties from the implicit restore triggered by /restore

### ISSUE THREAD

**Issue by baronfel** [dotnet/msbuild#9778]
### Summary

In https://github.com/dotnet/sdk/pull/38956 we had to jump through a bunch of hoops to navigate the situations where the SDK needs to trigger a 'separate restore' - meaning a Restore operation that doesn't include various properties that the user may want to be sent to the subsequent Build/Publish/etc invocation. The most common scenario here is `TargetFramework`.

Because TargetFramework negatively influences Restore, we remove it, perform a Restore, then perform the original call the user requested. This is both fragile, and it creates overhead in the build from running an entire separate MSBuild invocation.

There should be some mechanism to either use the existing `-restoreProperty` or a new flag to explicitly un-set properties for the Restore triggered by `/restore`, so that these call patterns could be accounted for by the engine itself, instead of callers having to patch it together.

### Background and Motivation

This most clearly surfaces when people try to use the new `-getProperty` family of flags in conjunction with a build call that includes TFMs.

### Proposed Feature

A new `-rrp/-removeRestoreProperty` flag that takes a list of properties to unset for a Restore call.

### Alternative Designs

`-rp` could be extended to allow unset via `-rp:Property=`, though this would be ambiguous with an explicit set to an empty value.

**Comment by rainersigwald** [comment]
> `-rp` could be extended to allow unset via `-rp:Property=`, though this would be ambiguous with an explicit set to an empty value.

This was my first instinct. I'm surprised we allow explicit set to an empty value, since that's not really expressible in a project (it's ambiguous whether it's set to empty or unset-and-thus-returns-empty).

**Comment by AArnott** [comment]
I hit this repeatedly for commands like `msbuild /r /p:Platform=x64`. I want to build x64 -- I don't want _restore_ to get that property though, as that causes restore to do the wrong things.

**Comment by baronfel** [comment]
@AArnott in this specific case, as soon as you provide _any_ restore-specific properties (via `-rp`), none of the `-p` properties are propogated to restore. So you can explicitly control the set of properties provided to restore by 'mirroring' all of the `-p` you need at restore-time to `-rp`s, and dropping the ones you don't care about.

**Comment by AArnott** [comment]
In my case, I don't have any global properties that I want restore to see. So it sounds like a workaround is to add `-rp:FakeProperty=1` to the command line?

**Comment by jeffkl** [comment]
yes, when you specify one or more `/restoreProperty` values, that set of properties are the only ones passed to restore.
